lib/capability/capability.go

```diff
@@ -15,8 +16,14 @@ const (
 )
 
 type ClusterCapabilities struct {
-	KnownCapabilities   map[configv1.ClusterVersionCapability]struct{}
-	EnabledCapabilities map[configv1.ClusterVersionCapability]struct{}
+	KnownCapabilities     map[configv1.ClusterVersionCapability]struct{}
+	EnabledCapabilities   map[configv1.ClusterVersionCapability]struct{}
+	TaintedInCapabilities []configv1.ClusterVersionCapability
+}
+
+func (c *ClusterCapabilities) Equal(capabilities *ClusterCapabilities) bool {
+	return reflect.DeepEqual(c.KnownCapabilities, capabilities.KnownCapabilities) &&
+		reflect.DeepEqual(c.EnabledCapabilities, capabilities.EnabledCapabilities)
```

Comment:
This is the only bit that matters for `equalSyncWork`, isn't it?  `KnownCapabilities` would only ever change as a result of a target-release change.  And even then, `KnownCapabilities` changes would only affect "why we're skipping this manifest" log messages.  I'm not strongly against having this `Equal` method, but I think all we need is this `reflect.DeepEqual(c.EnabledCapabilities, capabilities.EnabledCapabilities)` inlined in `equalSyncWork`